src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -0,0 +1,214 @@
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+
+public class ElasticSearchService implements ConsentLogger {
+
+  private final RestClient esClient;
+  private final ElasticSearchConfiguration esConfig;
+  private final DatasetDAO datasetDAO;
+  private final DataAccessRequestDAO dataAccessRequestDAO;
+  private final DacDAO dacDAO;
+  private final UserDAO userDAO;
+  private final UseRestrictionConverter useRestrictionConverter;
+
+  public ElasticSearchService(
+      RestClient esClient,
+      ElasticSearchConfiguration esConfig,
+      UseRestrictionConverter useRestrictionConverter,
+      DatasetDAO datasetDAO,
+      DataAccessRequestDAO dataAccessRequestDAO,
+      DacDAO dacDAO,
+      UserDAO userDAO) {
+    this.esClient = esClient;
+    this.esConfig = esConfig;
+    this.useRestrictionConverter = useRestrictionConverter;
+    this.datasetDAO = datasetDAO;
+    this.dataAccessRequestDAO = dataAccessRequestDAO;
+    this.dacDAO = dacDAO;
+    this.userDAO = userDAO;
+  }
+
+
+  private static final String bulkHeader = """
+      { "index": {"_type": "dataset", "_id": "%d"} }
+      """;
+
+  public Response indexDatasets(List<DatasetTerm> datasets) throws IOException {
+    List<String> bulkApiCall = new ArrayList<>();
+
+    datasets.forEach((dsTerm) -> {
+      bulkApiCall.add(bulkHeader.formatted(dsTerm.getDatasetId()));
+      bulkApiCall.add(GsonUtil.getInstance().toJson(dsTerm) + "\n");
+    });
+
+    Request bulkRequest = new Request(
+        "PUT",
+        "/" + esConfig.getDatasetIndexName() + "/_bulk");
+
+    bulkRequest.setEntity(new NStringEntity(
+        String.join("", bulkApiCall) + "\n",
+        ContentType.DEFAULT_BINARY));
+
+    return esClient.performRequest(bulkRequest);
+  }
+
+  public UserTerm toUserTerm(User user) {
+    if (Objects.isNull(user)) {
+      return null;
+    }
+
+    UserTerm term = new UserTerm();
+
+    term.setUserId(user.getUserId());
+
+    return term;
+  }
+
+  public StudyTerm toStudyTerm(Study study) {
+    if (Objects.isNull(study)) {
+      return null;
+    }
+
+    StudyTerm term = new StudyTerm();
+
+    term.setDescription(study.getDescription());
+    term.setStudyName(study.getName());
+    term.setStudyId(study.getStudyId());
+    term.setDataTypes(study.getDataTypes());
+    term.setPiName(study.getPiName());
+    term.setPublicVisibility(study.getPublicVisibility());
+
+    findStudyProperty(
+        study.getProperties(), "phenotypeIndication"
+    ).ifPresent(
+        prop -> term.setPhenotype((String) prop.getValue())
+    );
+
+    findStudyProperty(
+        study.getProperties(), "species"
+    ).ifPresent(
+        prop -> term.setSpecies((String) prop.getValue())
+    );
+
+    findStudyProperty(
+        study.getProperties(), "dataCustodianEmail"
+    ).ifPresent(
+        prop -> term.setDataCustodian((String) prop.getValue())
```

Comment:
This can be an array in some cases and throws an error. On dev, you'll see this with dataset ids 1422 -> 1436
```
Unable to map dataset id 1422 to a term: class com.google.gson.JsonArray cannot be cast to class java.lang.String (com.google.gson.JsonArray is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
```